src/sync.cpp

```diff
@@ -167,21 +156,22 @@ bool combinedFingerprint(LightFileFingerprint& ffp, const localnode_map& nodeMap
 }
 
 // Combines the fingerprints of all files in the given paths
-bool combinedFingerprint(LightFileFingerprint& ffp, FileSystemAccess& fsaccess, const set<string>& paths)
+bool combinedFingerprint(LightFileFingerprint& ffp, FileSystemAccess& fsaccess, const set<LocalPath>& paths)
 {
     bool success = false;
-    for (const auto& path : paths)
+    for (auto& path : paths)
     {
         auto fa = fsaccess.newfileaccess(false);
-        if (!fa->fopen(const_cast<string*>(&path), true, false))
+        auto pathArg = path; // todo: sort out const
+        if (!fa->fopen(pathArg, true, false))
```

Comment:
```suggestion
        if (!fa->fopen(const_cast<LocalPath&>(path), true, false))
```
? :thinking:

Reply:
This is one that will come right after the changes in the subsequent ticket, after which those parameters really can be const.   I think casting away const can be a bit dangerous so I'd rather avoid that, and wait for the proper fix?  thanks